sp/src/game/server/episodic/npc_advisor.cpp

```diff
@@ -1661,6 +1676,31 @@ bool CNPC_Advisor::HandleInteraction( int interactionType, void *data, CBaseComb
 
 	return BaseClass::HandleInteraction( interactionType, data, sourceEnt );
 }
+
+//=========================================================
+// TraceAttack is overridden here to handle shield impacts.
+//=========================================================
+void CNPC_Advisor::TraceAttack( const CTakeDamageInfo &info, const Vector &vecDir, trace_t *ptr, CDmgAccumulator *pAccumulator )
+{
+	m_fNoDamageDecal = false;
+
+	// Shield particle
+	if ( IsShieldOn() )
+	{
+		QAngle vecAngles;
+		VectorAngles( ptr->startpos - ptr->endpos, vecAngles );
+		DispatchParticleEffect( "warp_shield_impact_BMan", ptr->endpos, vecAngles );
```

Comment:
`trace_t` already contains the surface's normal. IIRC it's within either the `plane` or the `surface` member vars. I would recommend using that instead, as it's actually based on the impact surface relative to the model.